implementations/rust/ockam/ockam_transport_tcp/src/workers/listener.rs

```diff
@@ -38,6 +38,10 @@ impl TcpListenProcessor {
 impl Processor for TcpListenProcessor {
     type Context = Context;
 
+    async fn initialize(&mut self, ctx: &mut Context) -> Result<()> {
+        ctx.set_cluster(crate::CLUSTER_NAME).await
```

Comment:
There might be a race condition here when you create the TcpTransport and immediately shut down the node. Then the worker won't have initialised before it's cleaned up.
I'm not sure how much of a concert this is but we have two options I think:

1. Make `start_worker` wait for worker initialisation to complete
2. Pass cluster name via Context API (for example `start_cluster_worker(addr, cluster_name, worker)`

The first one obviously has other implications. The second one might be a bit verbose but probably otherwise safer.

Reply:
The first one will potentially hurt performance
The second one sounds good, I'll work on passing options to start_worker (it's needed by what I'm doing), maybe we'll be able to setup cluster as well